app/controllers/application_controller.rb

```diff
@@ -18,10 +18,10 @@ def sys_app_groups
     @sys_app_groups ||= OodAppGroup.groups_for(apps: SysRouter.apps)
   end
 
-  def set_announcement
-    path = Pathname.new(ENV["OOD_ANNOUNCEMENT_PATH"] || "/etc/ood/config/announcement.md")
-    @announcement = path.read if path.file?
+  def set_announcements
+    @announcements = Announcements.all
```

Comment:
A consideration.

1. The test for `Announcements.all` mocks `Configuration.announcement_path` because it knows that the implementation of  `Announcements.all` uses `Configuration.announcement_path`
2.  `Announcements.all` contains guard clauses to handle the case that `Configuration.announcement_path` is nil 


If you ensure `Configuration.announcement_path` never returns nil, you can get rid of the other guard clauses and test mock:

```ruby
+      [
 +        "/etc/ood/config/announcement.md",
 +        "/etc/ood/config/announcement.yml",
 +        "/etc/ood/config/announcements.d"
 +      ].map {|p| Pathname.new(p)}.detect(&:exist?) || Pathname.new("/etc/ood/config/announcement.md")
```


Then the code could be simplified by:

1. dropping `Announcement.all` and renaming `Announcement.parse(path)` to  `Announcement.all(path)`
2. change this line from `@announcements = Announcements.all` to `@announcements = Announcements.all(Configuration.announcement_path)`
3. remove the old for `all` that mocks `Configuration.announcement_path` 

Of course, `Announcements.all(Configuration.announcement_path)` is not as pretty as  `Announcements.all` but it does make  `Announcements.all` functional.

I also notice that `Configuration.announcement_path` could be "" if we set that value. I don't think that is a likely edge case, and even if that was the case, `Pathname.new("").expand_path` returns the current directory, which probably won't have an announcement file (and in fact, that could be a useful trick when developing; just set OOD_ANNOUNCEMENT_PATH to ""; maybe that should even be the default value in development so we can just drop those files in place).